src/main/java/com/cloudant/client/api/ClientBuilder.java

```diff
@@ -0,0 +1,526 @@
+        return this;
+    }
+
+    /**
+     * Set the maximum number of connections to maintain in the connection pool.
+     * <P>
+     * Note: this setting only applies if using the optional OkHttp dependency. If OkHttp is not
+     * present then the JVM configuration is used for pooling. Consult the JVM documentation for
+     * the {@code http.maxConnections} property for further details.
+     * </P>
+     * Defaults to {@link #DEFAULT_MAX_CONNECTIONS}
+     *
+     * @param maxConnections the maximum number of simultaneous connections to open to the server
+     * @return this ClientBuilder object for setting additional options
+     */
+    public ClientBuilder setMaxConnections(int maxConnections) {
+        this.maxConnections = maxConnections;
+        return this;
+    }
+
+    /**
+     * Sets a proxy url for the client connection.
+     *
+     * @param proxyURL the URL of the proxy server
+     * @return this ClientBuilder object for setting additional options
+     */
+    public ClientBuilder setProxyURL(URL proxyURL) {
+        this.proxyURL = proxyURL;
+        return this;
+    }
+
+    /**
+     * Sets an optional proxy username for the client connection.
+     *
+     * @param proxyUser username for the proxy server
+     * @return this ClientBuilder object for setting additional options
+     */
+    public ClientBuilder setProxyUser(String proxyUser) {
+        this.proxyUser = proxyUser;
+        return this;
+    }
+
+    /**
+     * Sets an optional proxy password for the proxy user specified by
+     * {@link #setProxyUser(String)}.
+     *
+     * @param proxyPassword password for the proxy server user
+     * @return this ClientBuilder object for setting additional options
+     */
+    public ClientBuilder setProxyPassword(String proxyPassword) {
+        this.proxyPassword = proxyPassword;
+        return this;
+    }
+
+    /**
+     * Flag to disable hostname verification and certificate chain validation. This is not
+     * recommended, but for example could be useful for testing with a self-signed certificate.
+     * <P>
+     * The default is {@code setSSLAuthenticationDisabled(false)} meaning that hostname
+     * verification and certificate chain validation is done using the JVM default settings.
+     * </P>
+     *
+     * @param disabled true to disable, defaults to false
+     * @return this ClientBuilder object for setting additional options
+     * @see #setAuthenticatedModeSSLSocketFactory(SSLSocketFactory)
+     */
+    public ClientBuilder setSSLAuthenticationDisabled(boolean disabled) {
+        this.isSSLAuthenticationDisabled = disabled;
+        return this;
+    }
+
+    /**
+     * Specifies the custom SSLSocketFactory to use when connecting to Cloudant over a
+     * <code>https</code> URL, when SSL authentication is enabled.
+     * <P>
+     * Note: this will have no effect if {@code setSSLAuthenticationDisabled(true)} is used to
+     * disable SSL authentication.
```

Comment:
The doc is correct, I was looking at the code yesterday, but I also did feel this was a bit counter intuitive. I wonder if we should actually check within the builder to ensure they are mutually exclusive and throw an exception, rather than assigning priority to one or the other.